apps/api/models/data_loader/PDF.py

```diff
@@ -0,0 +1,120 @@
+import io
+import logging
+
+from langchain.schema import Document
+from langchain.text_splitter import CharacterTextSplitter
+from models.base import dataset_manager
+from pdfminer.converter import TextConverter
+from pdfminer.layout import LAParams
+from pdfminer.pdfdocument import PDFDocument
+from pdfminer.pdfinterp import PDFPageInterpreter, PDFResourceManager
+from pdfminer.pdfpage import PDFPage
+from pdfminer.pdfparser import PDFParser
+from pydantic import BaseModel, Field
+from utils.StorageClient import GoogleCloudStorageClient
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class PDFSplitterOption(BaseModel):
+    type: str = Field(default="character")
+    chunk_size: int = Field(default=1000)
+    chunk_overlap: int = Field(default=0)
+
+
+class PDFEmbeddingOption(BaseModel):
+    model: str = Field(default="gpt-3.5-turbo")
+
+
+class PDFRetrivalOption(BaseModel):
+    splitter: PDFSplitterOption = Field(default_factory=PDFSplitterOption)
+    embedding: PDFEmbeddingOption = Field(default_factory=PDFEmbeddingOption)
+
+
+class PDFLoader:
+    @staticmethod
+    def load_and_split_documents(
```

Comment:
Since we are responsible for each search result, we have to provide the functionality of the data source, and the langchain wrapper splitter doesn't implement similar functionality.